generators/app/templates/src/app/components/CustomTextInput/index.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState, useCallback, memo } from "react";
+import { View, TextInput, Text } from "react-native";
+import PropTypes from "prop-types";
+import CustomText from "@components/CustomText";
+import { transparent } from "@constants/colors";
+
+import ShowPassword from "./components/ShowPassword";
+import { CustomTextInputProps } from "./interface";
+import styles from "./styles";
+
+const CustomTextInput = (props): CustomTextInputProps => {
+  const [showPassword, setShowPassword] = useState(false);
+
+  const handleShowPassword = useCallback(
+    () => setShowPassword((prevShowPassword) => !prevShowPassword),
+    []
+  );
+
+  const {
+    value,
+    placeholderTextColor,
+    title,
+    titleStyles,
+    multiline,
+    bottomBorder,
+    style,
+    onChange,
+    onBlur,
+    onFocus,
+    textStyles,
+    secureTextEntry,
+    showEye,
+    autoCompleteType,
+  } = props;
+
+  const placeholderColor = value ? transparent : placeholderTextColor;
+
+  return (
+    <>
+      {title && (
+        <CustomText gray small style={[styles.title, titleStyles]}>
+          {title}
+        </CustomText>
+      )}
+      <View
+        style={[
+          multiline ? styles.multilineContainer : styles.container,
+          bottomBorder && styles.bottomBorder,
+          style,
+        ]}
+      >
+        <TextInput
+          {...props}
+          allowFontScaling={false}
+          onChangeText={onChange}
+          onBlur={onBlur}
+          onFocus={onFocus}
+          value={value}
+          style={[
+            styles.inputStyle,
+            multiline ? styles.multilineInput : styles.singleInput,
+            textStyles,
+          ]}
+          placeholderTextColor={placeholderColor}
+          secureTextEntry={secureTextEntry && !showPassword}
+          autoCompleteType={secureTextEntry ? "off" : autoCompleteType}
+        />
+        {secureTextEntry && showEye && (
+          <ShowPassword
+            onShowPassword={handleShowPassword}
+            passwordVisible={showPassword}
+          />
+        )}
+      </View>
+    </>
+  );
+};
+
```

Comment:
Compare `CustomTextInput` and `AnimatedCustomTextInput` please. The only difference that there should be between them is the animated label but there are other differences such as this component doesn't have the `error` below the `TextInput` for example